### PR TITLE
Made a basic enhancment and fixed two bugs

### PR DESCRIPTION
**Pull request by JaysonHahn**
## Overview

Made a basic enhancement and fixed two bugs

## Changes Made

- Upcoming menus meal filter defaults based on time of day
    - During a meal time the filter defaults to that time
    -  Not during a specific meal time the filter defaults to the next meal time
- Fixed bug that eatery data would not populate (network request function not called)
- Made the report issues button able to be clicked from not just its label

## Test Coverage

Manual testing. Tested different times of day to ensure the meal data filter defaulted to correct meal.

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>
Bug Fixes

  
  

https://github.com/cuappdev/eatery-blue-ios/assets/46629787/299e37a2-7e7b-437d-a9cd-8c4b334b5589

</details>

### CODE REVIEW

File: Eatery Blue/UI/SettingsSupport/SettingsSupportView.swift [hunk 1]
```diff
@@ -95,11 +95,10 @@ struct SettingsSupportView: View {
                             .font(Font(UIFont.preferredFont(for: .body, weight: .semibold)))
                         Spacer()
                     }
-                }
+                })
```
Comment (vinnie4k) [inline Eatery Blue/UI/SettingsSupport/SettingsSupportView.swift:98]: Keep the trailing closure syntax. Undo these changes.

File: Eatery Blue/UI/MenusScreen/MenusModelController.swift [hunk 2]
```diff
@@ -29,7 +29,29 @@ class MenusModelController: MenusViewController {
...
+                    return "Dinner"
+                } else {
+                    return "Late Dinner"
+                }
+            } else {
+                return "Breakfast"
+            }
+        } else {
+            return "Breakfast"
+        }
```
Comment (vinnie4k) [inline Eatery Blue/UI/MenusScreen/MenusModelController.swift:53]: This should be filtered out to a `String+Extension` file with proper docstring comments for the function.

File: Eatery Blue/UI/MenusScreen/MenusModelController.swift [hunk 3]
```diff
@@ -29,7 +29,29 @@ class MenusModelController: MenusViewController {
     private lazy var loadCells: () = updateCellsFromState()
     
     private var selectedDay: Day = Day()
-    private var currentMealType: String = "Breakfast"
+    private var currentMealType: String = {
+        let date = Date()
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "HH"
+        if let hour : Int = Int(dateFormatter.string(from: date)) {
```
Comment (vinnie4k) [inline Eatery Blue/UI/MenusScreen/MenusModelController.swift:36]:
```suggestion
        if let hour = Int(dateFormatter.string(from: date)) {
```

File: Eatery Blue/UI/MenusScreen/MenusModelController.swift [hunk 4]
```diff
@@ -29,7 +29,29 @@ class MenusModelController: MenusViewController {
...
+            dateFormatter.dateFormat = "mm"
+            if let minute : Int = Int(dateFormatter.string(from: date)){
```
Comment (vinnie4k) [inline Eatery Blue/UI/MenusScreen/MenusModelController.swift:38]:
```suggestion
            if let minute = Int(dateFormatter.string(from: date)) {
```

File: Eatery Blue/UI/MenusScreen/MenusModelController.swift (hunk 2 above)
Comment (vinnie4k) [inline Eatery Blue/UI/MenusScreen/MenusModelController.swift:53]: Can you use a `guard let` instead to prevent "callback hell" code.

File: Eatery Blue/UI/EateryCard/EateryExpandableCardDetailView.swift [hunk 5]
```diff
@@ -130,11 +130,11 @@ class EateryExpandableCardDetailView: UIView {
     
     @objc private func didTapEateryDetails(_ sender: UITapGestureRecognizer) {
         if let navigationController = findNavigationController() {
+            guard let eatery = eatery else { return }
```
Comment (vinnie4k) [inline Eatery Blue/UI/EateryCard/EateryExpandableCardDetailView.swift:133]: Add new line below.

File: Eatery Blue/UI/EateryCard/EateryExpandableCardContentView.swift [hunk 6]
```diff
@@ -118,12 +118,11 @@ class EateryExpandableCardContentView: UIView {
     
     @objc private func didTapEateryDetails(_ sender: UITapGestureRecognizer) {
         if let navigationController = findNavigationController() {
+            guard let eatery = eatery else { return }
```
Comment (vinnie4k) [inline Eatery Blue/UI/EateryCard/EateryExpandableCardContentView.swift:121]: Add new line below.

File: Eatery Blue.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -7,7 +7,6 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
-		2E08E1952AD469BB0036E0E1 /* GoogleService-Info.plist in Resources */ = {isa = PBXBuildFile; fileRef = 2E08E1942AD469BB0036E0E1 /* GoogleService-Info.plist */; };
```
Comment (vinnie4k) [inline Eatery Blue.xcodeproj/project.pbxproj:10]: I will work with you and @antoinette-torres (maybe during work session) to prevent this from happening. You both will need to make sure that your branches don't have this issue.

File: Eatery Blue/UI/SettingsSupport/SettingsSupportView.swift [hunk 8]
```diff
@@ -95,11 +95,10 @@ struct SettingsSupportView: View {
...
                 .foregroundColor(.white)
                 .background(Color("EateryBlue"))
                 .clipShape(Capsule())
-                .buttonStyle(.plain)
```
Comment (vinnie4k) [inline Eatery Blue/UI/SettingsSupport/SettingsSupportView.swift:102]: Nice! Tapping on the empty space of a`Button` with a `.plain` style won't trigger the action. Sorry, I know you asked me to help about this but I was a bit busy with iOS course stuff. Good job figuring this out!

**Review by vinnie4k — CHANGES_REQUESTED** [review]
Great job! Just address my comments and request me again.

File: Eatery Blue.xcodeproj/project.pbxproj (hunk 7 above)
Comment (JaysonHahn) [reply Eatery Blue.xcodeproj/project.pbxproj:10]: yes usually i only stage the files that i changed code in but this time i kind of got lazy and did "add .". I'll be sure to get in the habit of that from now on.

**Review by vinnie4k — APPROVED** [review]
I pushed some styling changes to your code but everything looks good! Going to merge it right now so we can get this out to TestFlight ASAP.
